File: 171_Excel_Sheet_Column_Number.js
/**
 * @param {string} s
 * @return {number}
 */
var titleToNumber = function (s) {
    var l = s.length;
    var nums = s.split("");
    var ans = 0;
    nums.map(function(num, idx) {
        ans += Math.pow(26, l - idx - 1) * alphaToNum(num);
    });

    return ans;
};

/**
 * @param {string} a
 * @return {number}
 */
var alphaToNum = function (a) {
    if (a === "A") {
        return 1;
    } else {
        return a.charCodeAt(0) - "A".charCodeAt(0) + 1;
    }
}

File: 171_Excel_Sheet_Column_Number.c
#include <string.h>
#include <math.h>
int titleToNumber(char* s) {
  int ans = 0;
  int exp = 0;
  for (size_t i=strlen(s); i>0; i--) {
    int ascii_int = (int)s[i-1] - 64;
    int mutiply = (int)pow(26, exp++);
    ans += ascii_int * mutiply;
  }
  
  return ans;
}
